### PR TITLE
Add eager workflow start sample

### PR DESCRIPTION
**Pull request by antlai-temporal**
## What was changed

Add an example of starting a workflow in eager mode, a latency optimization.
## Why?

We are launching this feature in PreRelease soon

## Checklist

1. Closes 
#333

### CODE REVIEW

File: eager-workflow-start/main.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		log.Fatalln("Unable to start worker", err)
+	}
+	defer w.Stop()
+
+	workflowOptions := client.StartWorkflowOptions{
+		ID:        "eager_wf" + uuid.New(),
+		TaskQueue: taskQueueName,
+
+		// 3. Set this flag to true to enable EWS.
+		EnableEagerStart: true,
```
Comment (cretz) [inline eager-workflow-start/main.go:42]: Hrmm, wonder if it's a bit much to have an entire sample just to set true on a single workflow option (I wouldn't expect separate samples for every other start option). But I don't mind I guess if it makes sense generally.

Comment (Quinn-With-Two-Ns) [reply eager-workflow-start/main.go:42]: I think it is worth making a sample for since it also shows you have to reuse the client

**Review by Quinn-With-Two-Ns — APPROVED** [review]

File: eager-workflow-start/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+### Steps to run this sample:
```
Comment (Quinn-With-Two-Ns) [inline eager-workflow-start/README.md:1]: Can we add a short description here about eager workflow start and how the sample works like some of our other samples?

Comment (antlai-temporal) [reply eager-workflow-start/README.md:1]: Done

**Pull request merged**